design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,191 @@
+type Mutation {
+    addPurchaseOrderComment(input: AddPurchaseOrderCommentInput!): AddPurchaseOrderCommentOutput
+    approvePurchaseOrder(input: ApprovePurchaseOrderInput!): ApprovePurchaseOrderOutput
+    cancelPurchaseOrder(input: CancelPurchaseOrderInput!): CancelPurchaseOrderOutput
+    rejectPurchaseOrder(input: RejectPurchaseOrderInput!): RejectPurchaseOrderOutput
+}
+
+input ApprovePurchaseOrderInput {
+    purchase_order_uid: ID!
+}
+
+type ApprovePurchaseOrderOutput {
+    purchase_order: PurchaseOrder
+}
+
+input CancelPurchaseOrderInput {
+    purchase_order_uid: ID!
+}
+
+type CancelPurchaseOrderOutput {
+    purchase_order: PurchaseOrder
+}
+
+input RejectPurchaseOrderInput {
+    purchase_order_uid: ID!
+}
+
+type RejectPurchaseOrderOutput {
+    purchase_order: PurchaseOrder
+}
+
+input AddPurchaseOrderCommentInput {
+    purchase_order_uid: ID!
+    comment: String!
+}
+
+type AddPurchaseOrderCommentOutput {
+    purchase_order: PurchaseOrder
+    comment: PurchaseOrderComment
+}
+
+type Customer {
+    purchase_orders(filter: PurchaseOrdersFilterInput, currentPage: Int = 1, pageSize: Int = 20): PurchaseOrders
+    purchase_order(uid: ID!): PurchaseOrder
```

Comment:
Should these be on `Query`? I _believe_ managers can see their employees' purchase orders in B2B

Reply:
The idea behind putting everything under `Customer` is that visibility into purchase orders is different for different customers. And you are right that managers can see into their employees' purchase orders.

Additional minor point is that all purchase order functionality is currently part of customer's "My account".

@nrkapoor what do you think?

Reply:
@DrewML Agree with @paliarush. These objects should be linked like a real graph and we shouldn't have to create a separate query.